library/private/options_tools.h.in

```diff
@@ -10,7 +10,19 @@ namespace f3d
 {
 namespace options_tools
 {
-// TODO expose parse methods in options API ?
+
+static inline const std::string SPACE_CHARS = " \t\r\v\n";
+
+//----------------------------------------------------------------------------
+/**
+ * A std::string_view trim function
+ */
+std::string_view trim(std::string_view s)
+{
+  s.remove_prefix(std::min(s.find_first_not_of(SPACE_CHARS), s.size()));
+  s.remove_suffix(std::min(s.size() - s.find_last_not_of(SPACE_CHARS) - 1, s.size()));
+  return s;
+}
```

Comment:
thanks :)

Reply:
Ok, then, what about using `string_view` as argument (which is making sense to avoid a copy), but return an actual `string` instead of letting the caller do the conversion explicitly?
```cpp
std::string trim(std::string_view s)
{
  constexpr std::string_view spaces = " \t\r\v\n";
  s.remove_prefix(std::min(s.find_first_not_of(spaces), s.size()));
  s.remove_suffix(std::min(s.size() - s.find_last_not_of(spaces) - 1, s.size()));
  return s;
}